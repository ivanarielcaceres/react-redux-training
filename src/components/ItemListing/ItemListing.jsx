import React from 'react';
import PropTypes from 'prop-types';
import { FaEraser } from 'react-icons/fa';
import { ListGroup, ListGroupItem, Button } from 'reactstrap';

const ItemListing = ({ items, removeItem }) => (
  <ListGroup>
    {items.map(item => (
      <ListGroupItem key={item.key}>
        {item.name}
        <Button color="danger" onClick={removeItem.bind(this, item.key)}>
          <FaEraser />
        </Button>
      </ListGroupItem>
    ))}
  </ListGroup>
);

ItemListing.propTypes = {
  items: PropTypes.arrayOf(PropTypes.object),
  removeItem: PropTypes.func.isRequired
};

export default ItemListing;
