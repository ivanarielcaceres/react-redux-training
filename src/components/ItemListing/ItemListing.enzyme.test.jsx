import React from 'react';
import { shallow, mount } from 'enzyme';
import ItemListing from 'components/ItemListing/ItemListing';
import { ListGroupItem } from 'reactstrap';

/**
 * Smoke test
 * */
describe('<ItemListing />', () => {
  const items = [
    { key: 0, name: 'Item 1' },
    { key: 1, name: 'Item 2' },
    { key: 2, name: 'Item 3' }
  ];
  let wrapper;
  const mockRemoveItemfn = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <ItemListing items={items} removeItem={mockRemoveItemfn} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Render without crasing', () => {
    shallow(<ItemListing items={items} removeItem={mockRemoveItemfn} />);
  });

  it('List must show 3 Items', () => {
    expect(wrapper.find(ListGroupItem).length).toBe(3);
  });
});
