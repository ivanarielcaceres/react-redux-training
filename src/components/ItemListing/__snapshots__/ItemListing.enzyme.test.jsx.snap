// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemListing /> List must show 3 Items 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemListing
    items={
        Array [
            Object {
              "id": 0,
              "name": "Item 1",
            },
            Object {
              "id": 1,
              "name": "Item 2",
            },
            Object {
              "id": 2,
              "name": "Item 3",
            },
          ]
    }
    removeItem={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ListGroupItem
          tag="li"
>
          Item 1
          <Button
                    color="danger"
                    onClick={[Function]}
                    tag="button"
          >
                    <Unknown />
          </Button>
</ListGroupItem>,
        <ListGroupItem
          tag="li"
>
          Item 2
          <Button
                    color="danger"
                    onClick={[Function]}
                    tag="button"
          >
                    <Unknown />
          </Button>
</ListGroupItem>,
        <ListGroupItem
          tag="li"
>
          Item 3
          <Button
                    color="danger"
                    onClick={[Function]}
                    tag="button"
          >
                    <Unknown />
          </Button>
</ListGroupItem>,
      ],
      "tag": "ul",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "Item 1",
            <Button
              color="danger"
              onClick={[Function]}
              tag="button"
>
              <Unknown />
</Button>,
          ],
          "tag": "li",
        },
        "ref": null,
        "rendered": Array [
          "Item 1",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Unknown />,
              "color": "danger",
              "onClick": [Function],
              "tag": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "Item 2",
            <Button
              color="danger"
              onClick={[Function]}
              tag="button"
>
              <Unknown />
</Button>,
          ],
          "tag": "li",
        },
        "ref": null,
        "rendered": Array [
          "Item 2",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Unknown />,
              "color": "danger",
              "onClick": [Function],
              "tag": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "Item 3",
            <Button
              color="danger"
              onClick={[Function]}
              tag="button"
>
              <Unknown />
</Button>,
          ],
          "tag": "li",
        },
        "ref": null,
        "rendered": Array [
          "Item 3",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Unknown />,
              "color": "danger",
              "onClick": [Function],
              "tag": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ListGroupItem
            tag="li"
>
            Item 1
            <Button
                        color="danger"
                        onClick={[Function]}
                        tag="button"
            >
                        <Unknown />
            </Button>
</ListGroupItem>,
          <ListGroupItem
            tag="li"
>
            Item 2
            <Button
                        color="danger"
                        onClick={[Function]}
                        tag="button"
            >
                        <Unknown />
            </Button>
</ListGroupItem>,
          <ListGroupItem
            tag="li"
>
            Item 3
            <Button
                        color="danger"
                        onClick={[Function]}
                        tag="button"
            >
                        <Unknown />
            </Button>
</ListGroupItem>,
        ],
        "tag": "ul",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "Item 1",
              <Button
                color="danger"
                onClick={[Function]}
                tag="button"
>
                <Unknown />
</Button>,
            ],
            "tag": "li",
          },
          "ref": null,
          "rendered": Array [
            "Item 1",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Unknown />,
                "color": "danger",
                "onClick": [Function],
                "tag": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "Item 2",
              <Button
                color="danger"
                onClick={[Function]}
                tag="button"
>
                <Unknown />
</Button>,
            ],
            "tag": "li",
          },
          "ref": null,
          "rendered": Array [
            "Item 2",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Unknown />,
                "color": "danger",
                "onClick": [Function],
                "tag": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "Item 3",
              <Button
                color="danger"
                onClick={[Function]}
                tag="button"
>
                <Unknown />
</Button>,
            ],
            "tag": "li",
          },
          "ref": null,
          "rendered": Array [
            "Item 3",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Unknown />,
                "color": "danger",
                "onClick": [Function],
                "tag": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ItemListing /> Render without crasing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemListing
    items={
        Array [
            Object {
              "id": 0,
              "name": "Item 1",
            },
            Object {
              "id": 1,
              "name": "Item 2",
            },
            Object {
              "id": 2,
              "name": "Item 3",
            },
          ]
    }
    removeItem={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ListGroupItem
          tag="li"
>
          Item 1
          <Button
                    color="danger"
                    onClick={[Function]}
                    tag="button"
          >
                    <Unknown />
          </Button>
</ListGroupItem>,
        <ListGroupItem
          tag="li"
>
          Item 2
          <Button
                    color="danger"
                    onClick={[Function]}
                    tag="button"
          >
                    <Unknown />
          </Button>
</ListGroupItem>,
        <ListGroupItem
          tag="li"
>
          Item 3
          <Button
                    color="danger"
                    onClick={[Function]}
                    tag="button"
          >
                    <Unknown />
          </Button>
</ListGroupItem>,
      ],
      "tag": "ul",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "Item 1",
            <Button
              color="danger"
              onClick={[Function]}
              tag="button"
>
              <Unknown />
</Button>,
          ],
          "tag": "li",
        },
        "ref": null,
        "rendered": Array [
          "Item 1",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Unknown />,
              "color": "danger",
              "onClick": [Function],
              "tag": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "Item 2",
            <Button
              color="danger"
              onClick={[Function]}
              tag="button"
>
              <Unknown />
</Button>,
          ],
          "tag": "li",
        },
        "ref": null,
        "rendered": Array [
          "Item 2",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Unknown />,
              "color": "danger",
              "onClick": [Function],
              "tag": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "Item 3",
            <Button
              color="danger"
              onClick={[Function]}
              tag="button"
>
              <Unknown />
</Button>,
          ],
          "tag": "li",
        },
        "ref": null,
        "rendered": Array [
          "Item 3",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Unknown />,
              "color": "danger",
              "onClick": [Function],
              "tag": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ListGroupItem
            tag="li"
>
            Item 1
            <Button
                        color="danger"
                        onClick={[Function]}
                        tag="button"
            >
                        <Unknown />
            </Button>
</ListGroupItem>,
          <ListGroupItem
            tag="li"
>
            Item 2
            <Button
                        color="danger"
                        onClick={[Function]}
                        tag="button"
            >
                        <Unknown />
            </Button>
</ListGroupItem>,
          <ListGroupItem
            tag="li"
>
            Item 3
            <Button
                        color="danger"
                        onClick={[Function]}
                        tag="button"
            >
                        <Unknown />
            </Button>
</ListGroupItem>,
        ],
        "tag": "ul",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "Item 1",
              <Button
                color="danger"
                onClick={[Function]}
                tag="button"
>
                <Unknown />
</Button>,
            ],
            "tag": "li",
          },
          "ref": null,
          "rendered": Array [
            "Item 1",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Unknown />,
                "color": "danger",
                "onClick": [Function],
                "tag": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "Item 2",
              <Button
                color="danger"
                onClick={[Function]}
                tag="button"
>
                <Unknown />
</Button>,
            ],
            "tag": "li",
          },
          "ref": null,
          "rendered": Array [
            "Item 2",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Unknown />,
                "color": "danger",
                "onClick": [Function],
                "tag": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "Item 3",
              <Button
                color="danger"
                onClick={[Function]}
                tag="button"
>
                <Unknown />
</Button>,
            ],
            "tag": "li",
          },
          "ref": null,
          "rendered": Array [
            "Item 3",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Unknown />,
                "color": "danger",
                "onClick": [Function],
                "tag": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
